Research/research modell to use 8 (#26)
Paeti/IPNeuronaleNetze: docs/research_results.md

# Research Results

## About which neural networks fits best our needs and which tasks we'll face.

<p>
title: my research results<br>
author: Patrick Reckeweg<br>
date: 13.11.18
</p>

After doing some further research I think we focused to much on just building the network.
We even forgot to think/ talk about whether a single network architecture solves both of our
classification problems. [Here](https://talhassner.github.io/home/projects/cnn_agegender/CVPR2015_CNN_AgeGenderEstimation.pdf) I found an interesting

> paper, doing the same stuff we want to do. IN the paper is said that cnn's are best choice for face recognition.

Reviewer: ```suggestion
paper, doing the same stuff we want to do. In the paper is said that cnn's are the best choice for face recognition.

```

> Same is said [here](https://towardsdatascience.com/the-4-convolutional-neural-network-models-that-can-classify-your-fashion-images-9fe7f3e5399d).

Reviewer: ```suggestion
The same is said [here](https://towardsdatascience.com/the-4-convolutional-neural-network-models-that-can-classify-your-fashion-images-9fe7f3e5399d).
```

Both the article and the link provide a mass of links to do some further research on, but I hadn't the time for doing it yet.

> Reading this papers I noticed that building the network was in both cases one of the smaller project tasks.

Reviewer: ```suggestion
Reading this papers I noticed that building the network was in both cases, one of the smaller project tasks.
```

[This](https://www.researchgate.net/publication/283356929_DEX_Deep_EXpectation_of_Apparent_Age_from_a_Single_Image) paper confirmed my adoption.
Trying to guess how long it could take to build the network I read [this](https://hackernoon.com/learning-keras-by-implementing-vgg16-from-scratch-d036733f2d5)

> and I would say we should be able to build the network in not more then 3 weeks(a 10 hours/person per week).

Reviewer: ```suggestion
and I would say we should be able to build the network in not more than 3 weeks (10 hours/person per week).
```

- understanding what convolutional nn's exactly do
- image group:
  - which iput vgg-16 takes
  - how to split the datasets
  - how to prepare the images(rotate or zoom them for training => better generalization)
  - get datasets
- nn group:
  - build the network
  - get in touch with tensorboard
  - play arround with smaller nn's and learn how to train(all the stuff with graphs and optimization)

<p>
title: Alex Notizen<br>
author: Alex Wiens<br>
date: 13.11.18
</p>

VGG-16 erkennt zwar gut alter und Geschlecht, für weitere Features ist es aber vielleicht sinnvoll erstmal einen Gesichts-ausschnitt zu erhalten (eventuell auch mit Augen Position um je nachdem nachher einfacher die Augenfarbe zu erhalten).
Hierfür bietet sich ein einfaches NN basierend auf "Haar-Cascades" an (also relativ einfache Kontrast Analyse zwischen verschiedenen Positionen im Bild) welches von Intel zur Verfügung gestellt wird und direkt mit OpenCV ausgeliefert wird (allerdings als separate XML Datei)

- https://github.com/shantnu/FaceDetect/blob/master/face_detect.py
- https://realpython.com/face-detection-in-python-using-a-webcam/

__english version:__

> VGG-16 is good for age and gender recognition, but for further features it might be useful to get face detail first (possibly with eye position to get the eye color easier afterwards). For this a simple NN based on "hair cascades" (i.e. relatively simple contrast analysis between different positions in the image) which is provided by Intel and shipped directly with OpenCV (but as a separate XML file** is useful.

Reviewer 2: ```suggestion
VGG-16 is good for age and gender recognition, but for further features it might be useful to get a face/face detail cutout first (possibly with eye position to get the eye color easier afterwards). For this a simple NN based on "Haar cascades" (i.e. relatively simple contrast analysis between different positions in the image) which is provided by Intel and shipped directly with OpenCV (but as a separate XML file** is useful.
```
